ocp_resources/pipelineruns.py

```diff
@@ -0,0 +1,62 @@
+# API reference: https://tekton.dev/docs/pipelines/pipelineruns/
+
+from ocp_resources.resource import NamespacedResource
+
+
+class PipelineRun(NamespacedResource):
+    api_group = NamespacedResource.ApiGroup.TEKTON_DEV
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        pipelineref=None,
+        params=None,
+        service_account_name=None,
+        client=None,
+        yaml_file=None,
+        **kwargs,
+    ):
+        """
+        Args:
+            name (str): Name of the pipelinerun
+            namespace (str): Namespace of the pipelinerun
+            pipelineref (str): Mandatory: Base pipeline to run pipelineruns
+            client: (DynamicClient): DynamicClient to use.
+            params (dict): Optional params to add during triggering a run.
+            params can be set/changed based on pipelineref.
+            example : params={"key1":"value1", "key2":"value2"}
+            Refer : https://tekton.dev/docs/pipelines/pipelineruns/#specifying-parameters
+            service_account_name (str): Optional to provide service account
+        """
+        if not pipelineref:
```

Comment:
`        if not pipelineref:` please move it inside `to_dict()`

Reply:
@myakove sure i can move it. The only reason i keep it there because if that is missing we don't wanted to proceed at all. I think it will save time that will go in assignmet and other checks.wdyt

Reply:
if this already exists, no need to pass it.
should be in to_dict

Reply:
Moved to to_dict()